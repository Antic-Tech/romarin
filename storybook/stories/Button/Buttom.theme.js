export default {
	buttonBorderRadius: 10
};
