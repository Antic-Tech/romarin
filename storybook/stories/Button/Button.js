import React from 'react';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import design from '../../theme';

const styles = StyleSheet.create({
	button: {
		borderRadius: design.borderRadius,
		padding: design.paddingMedium,
		backgroundColor: design.colorPrimary
	},
	label: {
		fontSize: design.fontSizeMedium,
		color: design.colorWhite
	}
});

const Button = props => (
	<TouchableOpacity {...props} style={styles.button} onPress={props.disabled ? undefined : props.onPress}>
		<Text style={styles.label}>{props.label}</Text>
	</TouchableOpacity>
);

Button.propTypes = {
	onPress: PropTypes.func,
	disabled: PropTypes.bool,
	full: PropTypes.bool,
	text: PropTypes.string
};

Button.defaultProps = {
	onPress: () => {}
};

export { Button as default };
