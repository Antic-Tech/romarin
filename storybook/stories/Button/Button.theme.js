export default core => ({
	buttonBorderRadius: core.coreBorderRadius,
	buttonPadding: core.corePaddingMedium,
	buttonTextColor: core.colorWhite,
	iconSizeXLarge: 52,
	iconSizeLarge: 44,
	iconSizeMedium: 36,
	iconSizeSmall: 28,
	iconSizeXSmall: 20
});
