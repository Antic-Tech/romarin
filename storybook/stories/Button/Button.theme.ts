import { BaseTheme } from '../../definitions/Base.theme';

export default (core: BaseTheme) => ({
	borderRadius: core.borderRadius,
	paddingHorizontal: core.paddingMedium,
	paddingVertical: core.paddingMedium,
	labelColor: core.colorWhite,
	labelWeight: '500',
	iconSizeXLarge: 52,
	iconSizeLarge: 44,
	iconSizeMedium: 36,
	iconSizeSmall: 28,
	iconSizeXSmall: 20
});
