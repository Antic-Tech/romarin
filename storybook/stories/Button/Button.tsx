import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet } from 'react-native';
import { Text, Touchable } from '../../components';
import ThemeProvider from '../../theme';

const Button = (props) => {
	const styles = StyleSheet.create({
		button: {
			borderRadius: ThemeProvider.theme.button.borderRadius,
			paddingHorizontal: ThemeProvider.theme.button.paddingHorizontal,
			paddingVertical: ThemeProvider.theme.button.paddingVertical,
			backgroundColor: ThemeProvider.theme.base.colorPrimary
		},
		label: {
			textAlign: 'center',
			fontWeight: ThemeProvider.theme.button.labelWeight
		}
	});

	const { onPress, disabled, full, label } = props;

	return (
		<Touchable type="opacity" style={styles.button} onPress={disabled ? undefined : onPress}>
			<Text style={styles.label} color={ThemeProvider.theme.button.labelColor}>
				{label}
			</Text>
		</Touchable>
	);
};

Button.propTypes = {
	onPress: PropTypes.func,
	disabled: PropTypes.bool,
	full: PropTypes.bool,
	label: PropTypes.string
};

Button.defaultProps = {
	onPress: () => {}
};

export { Button as default };

