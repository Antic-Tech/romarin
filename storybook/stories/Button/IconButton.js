import PropTypes from 'prop-types';
import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { MaterialIcons } from 'react-native-vector-icons';

import theme from '../../theme';

const styles = StyleSheet.create({
	button: {
		margin: theme.marginSmall,
		shadowColor: 'rgba(0,0,0,.4)',
		shadowOffset: { height: 1, width: 1 },
		shadowOpacity: 1,
		shadowRadius: 1,
		alignItems: 'center',
		justifyContent: 'center'
	},
	icon: {
		backgroundColor: 'transparent'
	}
});

const IconButton = props => {
	const buttonSizes = {
		xlarge: theme.buttonSizeXLarge,
		large: theme.buttonSizeLarge,
		medium: theme.buttonSizeMedium,
		small: theme.buttonSizeSmall,
		xsmall: theme.buttonSizeXSmall
	};
	const iconSizes = {
		xlarge: theme.iconSizeXLarge,
		large: theme.iconSizeLarge,
		medium: theme.iconSizeMedium,
		small: theme.iconSizeSmall,
		xsmall: theme.iconSizeXSmall
	};
	const { name, size, color, onPress, backgroundColor } = props;

	return (
		<View>
			<TouchableOpacity
				style={[
					styles.button,
					{
						borderRadius: buttonSizes[size],
						height: buttonSizes[size],
						width: buttonSizes[size],
						backgroundColor: theme[backgroundColor]
					}
				]}
				onPress={onPress}
			>
				<MaterialIcons style={styles.icon} size={iconSizes[size]} name={name} color={theme[color]} />
			</TouchableOpacity>
		</View>
	);
};

IconButton.propTypes = {
	name: PropTypes.string,
	size: PropTypes.string,
	color: PropTypes.string,
	backgroundColor: PropTypes.string,
	onPress: PropTypes.func
};

IconButton.defaultProps = {
	name: 'android',
	size: 'medium',
	color: 'colorWhite',
	backgroundColor: 'colorPrimary',
	onPress: () => {}
};

export { IconButton as default };
