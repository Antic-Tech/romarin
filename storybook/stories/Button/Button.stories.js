import React from 'react';
import { Text } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';

import Button from './Button';
import CenterView from '../CenterView';

storiesOf('Button', module)
	.addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
	.add('with text', () => <Button onPress={action('clicked-text')} label="Button" />)
	.add('with some emoji', () => (
		<Button onPress={action('clicked-emoji')}>
			<Text>😀 😎 👍 💯</Text>
		</Button>
	));
