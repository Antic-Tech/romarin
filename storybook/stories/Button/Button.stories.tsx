import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { Text } from 'react-native';
import { Page } from '../../components';
import ThemeProvider from '../../theme';
import Button from './Button';

storiesOf('Button', module).addDecorator((getStory) => <Page padded>{getStory()}</Page>).add('basic button', () => (
	<React.Fragment>
		<Text>{ThemeProvider.theme.colorPrimary}</Text>
		<Button onPress={()=>{}} label="Button" />
		<Text>label Emoji</Text>
		<Button onPress={action('clicked-emoji')} label="😀 😎 👍 💯" />
	</React.Fragment>
));
