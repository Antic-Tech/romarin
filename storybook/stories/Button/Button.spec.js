import React from 'react';
import Button from './Button';

import { verifySnapshot } from '../../utils/testUtils';

describe('Button', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<Button label="Hello Button"/>);
		});
	});
});
