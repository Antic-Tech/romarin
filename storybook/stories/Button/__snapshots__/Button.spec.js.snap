// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2C7CF6",
      "borderRadius": 10,
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 17,
      }
    }
  >
    Hello Button
  </Text>
</View>
`;
